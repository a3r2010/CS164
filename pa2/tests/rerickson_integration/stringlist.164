def newList() {
	def length = 0
	def list = "|"
	def append(item) {
	  list = list + item + "|"
	  length = length + 1 
	}
	def to_s() {
	  list
	}
	def clear() {
	  list = "|"
	  length = 0
	}
	lambda(func) {
	  if(func == "to_s") {
	    to_s()
	  } else {
	    if(func == "append") {
	      lambda(item) {
	        append(item)
	      }   
	    } else {
	      if(func == "length") {
	        length
	      } else {
	        if(func == "clear") {
	          clear()
	        }
	      }
	    }
	  }
	}
}

def list = newList()
list("append")(4)
list("append")("f")
list("append")("foo")
list("append")("Getting to 35 points is a huge f&#*ing challenge!")
print list("to_s")
print "My list has " + list("length") + ' items in it!'
list("clear")
print list("to_s")
print "My list has " + list("length") + ' items in it!'
