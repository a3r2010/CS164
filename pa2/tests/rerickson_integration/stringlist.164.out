|4|f|foo|Getting to 35 points is a huge f&#*ing challenge!|
My list has 4 items in it!
|
My list has 0 items in it!
